bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1073,7 +1073,7 @@ private static boolean checkObjectEquivalency(BType sourceType, BObjectType targ
                                     .map(BPackage::getName)
                                     .orElse(""),
                             Optional.ofNullable(rhsFunc.type.getPackage()).map(BPackage::getName).orElse(""),
-                            lhsFunc.flags, rhsFunc.flags)) {
+                            lhsFunc.flags, rhsFunc.flags) || !isSameMethodType(lhsFunc.flags, rhsFunc.flags)) {
                 return false;
```

Comment:
Shall we refactor this block a bit?

```java
        for (AttachedFunction lhsFunc : targetFuncs) {
            AttachedFunction rhsFunc = getMatchingInvokableType(sourceFuncs, lhsFunc, unresolvedTypes);

            if (rhsFunc == null) {
                return false;
            }

            if (!isInSameVisibilityRegion(...))  {
                return false;
            }

            if (Flags.isFlagOn(lhsFunc.flags, Flags.REMOTE) && !Flags.isFlagOn(rhsFunc.flags, Flags.REMOTE)) {
                return false;
            }
        }
```

Reply:
I've removed the separate method and moved the logic here since the current method name doesn't seem quite accurate, it's only used here and it's a simple check.

Reply:
Also, shouldn't the check be for something like 

`Flags.isFlagOn(lhsFunc.flags, Flags.REMOTE) != Flags.isFlagOn(rhsFunc.flags, Flags.REMOTE)` ?

If the RHS's function has remote but the LHS's doesn't shouldn't that be an error? @hasithaa

Reply:
Changed logic to check whether the remote qualifier is available or not with both functions of LHS and RHS as mentioned.